valve-monitor/web-app/src/services/prisma-datastore/PrismaDataProvider.ts

```diff
@@ -118,9 +132,18 @@ export class PrismaDataProvider implements DataProvider {
     // fetch all devices by project
     const devices = await this.getDevices();
 
-    // fetch most most recent data.qo event for each device
+    // fetch the most recent data.qo event for each device
     const deviceEvents = await Promise.all(
-      devices.map((device) => this.getLatestDeviceEvent(device.id)).flat()
+      devices
+        .map((device) => this.getLatestDeviceEvent(device.id, "data.qo"))
+        .flat()
+    );
+
+    // fetch the most recent data.qi event for each device
+    const controlEvents = await Promise.all(
```

Comment:
I would test if there are no `data.qi` events just to see how it handles / doesn't handle that scenario (I've been bitten by it more than once while building this app so far)